ReSharper.FSharp/src/FSharp.Psi/src/Util/XmlDocTemplateUtil.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using JetBrains.Annotations;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Impl;
+using JetBrains.ReSharper.Plugins.FSharp.Psi.Tree;
+using JetBrains.ReSharper.Psi.ExtensionsAPI;
+using JetBrains.ReSharper.Psi.Tree;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Util
+{
+  public static class XmlDocTemplateUtil
+  {
+    [NotNull]
+    public static string GetDocTemplate(IDocCommentBlock docCommentBlock, Func<int, string> linePrefix, string lineSeparator)
+    {
+      var text = new StringBuilder();
+      var line = 1;
+
+      text.Append($"summary>{lineSeparator}");
+      text.Append($"{linePrefix(line++)}{lineSeparator}");
+      text.Append($"{linePrefix(line++)}</summary>{lineSeparator}");
+
+      foreach (var parameter in GetParameters(docCommentBlock.Parent))
+        text.Append($"{linePrefix(line++)}<param name=\"{parameter}\"></param>{lineSeparator}");
+
+      return text.ToString();
+    }
+
+    private static IEnumerable<string> GetParameters(ITreeNode declaration) =>
```

Comment:
Why `declaration` is not `IDeclaration`?

Reply:
IBinding, for example, does not implement IDeclaration

Reply:
Maybe I will rename parameter name in the future